src/worker.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env python
+
+import os
+import json
+import traceback
+import logging
+import subprocess
+import configparser
+
+from connection import Connection
+
+conn = Connection()
+
+logging.basicConfig(
+    format="%(asctime)-15s [%(levelname)s] %(message)s",
+    level=logging.DEBUG,
+)
+
+config = configparser.RawConfigParser()
+config.read([
+    'worker.cfg',
+    '/etc/py_gpac_worker/worker.cfg'
+])
+
+class GPAC_worker:
+
+    def get_parameter(self, key, param):
+        key = "GPAC_" + key
+        if key in os.environ:
+            return os.environ.get(key)
+
+        return config.get('gpac', param)
+
+    def load_configuration(self):
+        self.gpac_bin_path = self.get_parameter('BIN_PATH', 'binpath')
+        self.gpac_lib_path = self.get_parameter('LIB_PATH', 'libpath')
+        self.mp4box_path = os.path.join(self.gpac_bin_path, "MP4Box")
+        self.env = os.environ.copy()
+        self.env["LD_LIBRARY_PATH"] = self.gpac_lib_path
+
+    def process(self, src_path, dst_path, options: dict):
+
+        command = [self.mp4box_path]
+        for key, value in options.items():
+            command.append(key)
+            command.append(str(value))
+        command.append("-out")
+        command.append(dst_path)
+        command.append(src_path)
```

Comment:
is it possible to print the command for debugging purpose ?

Reply:
Of course! Check: af1e749!